lib/compiler-llvm/src/translator/code.rs

```diff
@@ -303,15 +303,11 @@ impl FuncTranslator {
             FUNCTION_SECTION,
             RelocationTarget::LocalFunc(*local_func_index),
             |name: &String| {
-                Ok(
-                    if let Some(Symbol::LocalFunction(local_func_index)) =
-                        symbol_registry.name_to_symbol(name)
-                    {
-                        Some(RelocationTarget::LocalFunc(local_func_index))
-                    } else {
-                        None
+                Ok(symbol_registry.name_to_symbol(name).map(
+                    |Symbol::LocalFunction(local_func_index)| {
+                        RelocationTarget::LocalFunc(local_func_index)
```

Comment:
Huh, I didn't know you could pattern match in `map` like this and have it work 🤔

Reply:
Huh? I missed this one. Are we sure about this rewrite?

Reply:
I think if it compiles then it's probably correct. It looks like something that may not compile to me because `map` isn't exhaustive, but if it is, then I think it'll probably do what we want